core/metadata-3.2.md

```diff
@@ -20,26 +40,139 @@ This proposal aims to codify one mechanism for working with metadata: metadata
 may be configured through a client-to-server event created specifically for
 this purpose.
 
-## METADATA
+## Mechanisms
+
+This document defines the following new protocol features:
+
+* Capability: `metadata`
+* Server notification: `METADATA`
+* Client command: `METADATA`
+* Server reply and error numerics
+
+### Capability
+
+The `metadata` capability and value indicates that a server supports metadata, and communicates any configuration value keys that may affect the level of support.
+
+The ABNF format of the `metadata` capability and value is: (TODO verify ABNF)
+
+    capability ::= 'metadata' ['=' value]
+    value      ::= value_item [',' value_item]*
+    value_item ::= ( 'maxsub=' integer ) |  ( 'maxkey=' integer )
+
+The value keys are defined as follows:
+
+* `maxsub`: the maximum number of keys a client is allowed in its subscripion list. See the [Metadata subscriptions](#todo) section for more details.
+* `maxkey`: the maximum number of keys a client is allowed to set on its own nickname.
+
+
+TODO is this still required? include it for back compat?
+If `METADATA` is supported, it MUST be specified in `RPL_ISUPPORT` using the
+`METADATA` key. Servers MAY specify a limit on the number of explicitly-set
+keys per-user; the format in that case MUST be `METADATA=<integer>`, where
+`<integer>` is the limit.
+
+
+### Server notification
```

Comment:
"Server notification" is a phrase I've never seen before in any IRC docs or specs, and it honestly sounds pretty confusing. Can we change how this is referred to, for example 'Server message' or something else that's more familiar?

Reply:
I'm just a mere IRC user that wants progress over the IRC protocol. I'm just happy to see research and development over it, I consider it needs to adapt and improve to be relevant and be able to regain user base again against very popular but proprietary chat (+voip +easy image and file sharing, +server logging, +integration with other services such as GitLab and GitHub and more, +extremely easy to use and popular clients in major platforms to the point of being foolproof...) SaaS solutions. 

Anyway, I'm very curious and usually read a lot of documentation and books, including many stuff related to IRC and even it's related IETF RCF documents. 

From my very extremely limited Point Of View: I consider your argumentation seems reasonable to me. Please take it with a very tiny grain of salt, anyway.

Thanks to everyone for improving IRCv3×! I hope many great ideas get approved and implemented by mostly all IRC/chat clients/bouncers/matrix/psyc2-gnunet etc